nautobot/dcim/filters.py

```diff
@@ -159,21 +170,85 @@ class SiteFilterSet(NautobotFilterSet, TenancyFilterSet, StatusModelFilterSetMix
         to_field_name="slug",
         label="Region (slug)",
     )
+    has_circuit_terminations = RelatedMembershipBooleanFilter(
+        field_name="circuit_terminations",
+        label="Has circuit terminations",
+    )
+    has_devices = RelatedMembershipBooleanFilter(
+        field_name="devices",
+        label="Has devices",
+    )
+    powerpanels = django_filters.ModelMultipleChoiceFilter(
+        field_name="powerpanel",
+        queryset=PowerPanel.objects.all(),
+        label="Power panels",
+    )
```

Comment:
Ignorant question: why do we need the `powerpanels` filter explicitly defined, but not `circuit_terminations` or `devices`? Is it because we're missing a `related_name` on the `site` ForeignKey definition under the PowerPanel model?

Reply:
The field name is `powerpanel` and we wanted to call the filter `powerpanels`

Reply:
> (Aside: I wish we were consistent on when words get run together and when they don't - why `circuit_terminations` and `rack_groups` but `powerfeeds` and `powerpanels`?)

We should probably discuss this before I get too far on naming new filters. DeviceFilterSet is a good example where we have filters for:
- has_console_ports
- has_console_server_ports
- similarly named filters with underscores between each word

But Jathan's [gist](https://gist.github.com/jathanism/7c585954da4cb1a54997be31991cdaac#devicefilterset) calls out adding these as new filters with the names collapsed:
- consoleports
- has_consoleports
- consoleserverports
- has_consoleserverports

Maybe it's a backwards compatibility issue. `console_ports` doesn't actually filter on the console port objects, it just points to `has_console_ports`. So we need a new filter to actually filter on console ports but need to keep the functionality of the existing filter the same. Fortunately I haven't seen a lot of filtersets like this.

```py
    has_console_ports = RelatedMembershipBooleanFilter(
        field_name="consoleports",
        label="Has console ports",
    )
    console_ports = has_console_ports
```

Here's my opinion on standardizing the names based on the filter work I've done so far:
- use underscores between words in filter names regardless of the model name
- make an exception for templates? (`consoleserverports_templates`)
- filters that use a MultipleChoiceFilter or ModelMultipleChoiceFilter should be plural (`racks`, not `rack`) - or just make it easier and make everything plural unless it doesn't make sense for a specific field

Reply:
As discussed today we're going to use the model names for filter names. If the model name is incorrect, that will have to be fixed in a separate effort.